src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt

```diff
@@ -98,7 +113,38 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
 
         assertFalse("Medium severity IaC results should NOT be shown after filtering", isMediumSeverityShown())
 
-        // restore modified Registry value
-        isIacEnabledRegistryValue.setValue(isIacEnabledOldValue)
+        tearDownIacTest()
+    }
+
+    @Test
+    fun testIacErrorShown() {
+        // pre-test setup
+        setUpIacTest()
+
+        // mock IaC results
+        val iacError = SnykError("fake error", "fake path")
+        val iacResultWithError = IacResult(null, iacError)
+
+        mockkStatic("io.snyk.plugin.UtilsKt")
+        every { getIacService(project).isCliInstalled() } returns true
+        every { getIacService(project).scan() } returns iacResultWithError
+
+        // actual test run
+        project.service<SnykTaskQueueService>().scan()
+
+        val toolWindowPanel = project.service<SnykToolWindowPanel>()
+        PlatformTestUtil.waitWhileBusy(toolWindowPanel.getTree())
+
+        assertEquals(iacError, toolWindowPanel.currentIacError)
```

Comment:
Consider to use verify { ... } to make sure that a service call is actually happening and the description panel is populated by the result of the call, not from any other error.

Reply:
Here we check intermediate condition required for the test of the test to have sense.
Also I don't want to chain the test to any particulate service or method call. But instead would like to check that we have required final result: error shown in description panel.
PS But Thanks for clarification! will add check for the expected error text and not any random error.